Update Plank to Swift 3.1

maicki opened the pull request:
The biggest change is changing variable names from 1 char to > 3 certain and enum cases to lowercase. Especially the later leads to interesting problems as in a couple of places we used uppercase Swift keywords. I think it's open for discussion what we would like to do in this case. The cases are the following:

* `Schema.Enum` -> `Schema.enumT`
* `Root.Struct` -> `Root.structDecl`
* `Root.Enum` -> `Root.enumDecl`
* `Root.Class` -> `Root.classDecl`
* `MethodVisibility.Private` -> `MethodVisibility.privateM`
* `MethodVisibility.Public` -> `MethodVisibility.publicM`

(T = type, Decl = declaration and M = modifier)

Adresses: #38

pinterest-ios commented:
Build failure:
https://gist.githubusercontent.com/pinterest-ios/c2488ae00d6e69491a5ad2b0508a32e7/raw/cf121667cba2b61c66ac4a2d6b0301fb5a3f1a43/results.txt

bkase commented:
Also I like the T, Decl, and M suffixes

rahul-malik commented on Sources/Core/ObjectiveCIR.swift:
```diff
@@ -170,17 +170,17 @@ public struct ObjCIR {
-        case Case(condition: String, body: () -> [String])
-        case Default(body: () -> [String])
+        case caseStm(condition: String, body: () -> [String])
+        case defaultStm(body: () -> [String])
```
Can you changed these suffixes to "Stmt" instead of "Stm"?

rahul-malik commented:
Also can you update the .swift-version file to specify 3.1 and build the docker image ?

maicki commented:
@rahul-malik Addressed comments, updated `.swift-version` and tried building a docker image:

```
$ docker build -t plank .
Sending build context to Docker daemon  82.7 MB
Step 1/8 : FROM kylef/swiftenv:latest
 ---> 9761ef4985b1
Step 2/8 : MAINTAINER Pinterest
 ---> Using cache
 ---> cfaea03dd4db
Step 3/8 : RUN swiftenv install 3.1
 ---> Running in 591106ce2902
Downloading https://swift.org/builds/swift-3.1-release/ubuntu1604/swift-3.1-RELEASE/swift-3.1-RELEASE-ubuntu16.04.tar.gz
3.1 has been installed.
 ---> 1f2992912f14
Removing intermediate container 591106ce2902
Step 4/8 : RUN apt-get update &&     apt-get install -y --no-install-recommends         vim clang libicu-dev libcurl4-openssl-dev libssl-dev
 ---> Running in f3fb98b35adf
Hit:1 http://archive.ubuntu.com/ubuntu xenial InRelease
Get:2 http://archive.ubuntu.com/ubuntu xenial-updates InRelease [102 kB]
Get:3 http://archive.ubuntu.com/ubuntu xenial-security InRelease [102 kB]
Get:4 http://archive.ubuntu.com/ubuntu xenial-updates/main Sources [304 kB]
Get:5 http://archive.ubuntu.com/ubuntu xenial-updates/restricted Sources [3202 B]
Get:6 http://archive.ubuntu.com/ubuntu xenial-updates/universe Sources [188 kB]
Get:7 http://archive.ubuntu.com/ubuntu xenial-updates/main amd64 Packages [655 kB]
Get:8 http://archive.ubuntu.com/ubuntu xenial-updates/restricted amd64 Packages [13.2 kB]
Get:9 http://archive.ubuntu.com/ubuntu xenial-updates/universe amd64 Packages [578 kB]
Get:10 http://archive.ubuntu.com/ubuntu xenial-security/main Sources [82.2 kB]
Get:11 http://archive.ubuntu.com/ubuntu xenial-security/restricted Sources [2779 B]
Get:12 http://archive.ubuntu.com/ubuntu xenial-security/universe Sources [29.7 kB]
Get:13 http://archive.ubuntu.com/ubuntu xenial-security/main amd64 Packages [311 kB]
Get:14 http://archive.ubuntu.com/ubuntu xenial-security/restricted amd64 Packages [12.8 kB]
Get:15 http://archive.ubuntu.com/ubuntu xenial-security/universe amd64 Packages [132 kB]
Fetched 2517 kB in 2s (1240 kB/s)
Reading package lists...
Reading package lists...
Building dependency tree...
Reading state information...
libcurl4-openssl-dev is already the newest version (7.47.0-1ubuntu2.2).
clang is already the newest version (1:3.8-33ubuntu3.1).
The following additional packages will be installed:
  icu-devtools libgpm2 libicu55 libpython3.5 libssl1.0.0 vim-common
  vim-runtime zlib1g-dev
Suggested packages:
  gpm icu-doc ctags vim-doc vim-scripts vim-gnome-py2 | vim-gtk-py2
  | vim-gtk3-py2 | vim-athena-py2 | vim-nox-py2
Recommended packages:
  libssl-doc
The following NEW packages will be installed:
  icu-devtools libgpm2 libicu-dev libpython3.5 libssl-dev vim vim-common
  vim-runtime zlib1g-dev
The following packages will be upgraded:
  libicu55 libssl1.0.0
2 upgraded, 9 newly installed, 0 to remove and 45 not upgraded.
Need to get 26.7 MB of archives.
After this operation, 84.2 MB of additional disk space will be used.
Get:1 http://archive.ubuntu.com/ubuntu xenial/main amd64 libgpm2 amd64 1.20.4-6.1 [16.5 kB]
Get:2 http://archive.ubuntu.com/ubuntu xenial-updates/main amd64 libssl1.0.0 amd64 1.0.2g-1ubuntu4.6 [1082 kB]
Get:3 http://archive.ubuntu.com/ubuntu xenial-updates/main amd64 vim-common amd64 2:7.4.1689-3ubuntu1.2 [103 kB]
Get:4 http://archive.ubuntu.com/ubuntu xenial-updates/main amd64 libicu55 amd64 55.1-7ubuntu0.1 [7654 kB]
Get:5 http://archive.ubuntu.com/ubuntu xenial-updates/main amd64 icu-devtools amd64 55.1-7ubuntu0.1 [166 kB]
Get:6 http://archive.ubuntu.com/ubuntu xenial-updates/main amd64 libicu-dev amd64 55.1-7ubuntu0.1 [8561 kB]
Get:7 http://archive.ubuntu.com/ubuntu xenial-updates/main amd64 libpython3.5 amd64 3.5.2-2ubuntu0~16.04.1 [1360 kB]
Get:8 http://archive.ubuntu.com/ubuntu xenial/main amd64 zlib1g-dev amd64 1:1.2.8.dfsg-2ubuntu4 [168 kB]
Get:9 http://archive.ubuntu.com/ubuntu xenial-updates/main amd64 libssl-dev amd64 1.0.2g-1ubuntu4.6 [1344 kB]
Get:10 http://archive.ubuntu.com/ubuntu xenial-updates/main amd64 vim-runtime all 2:7.4.1689-3ubuntu1.2 [5164 kB]
Get:11 http://archive.ubuntu.com/ubuntu xenial-updates/main amd64 vim amd64 2:7.4.1689-3ubuntu1.2 [1036 kB]
debconf: delaying package configuration, since apt-utils is not installed
Fetched 26.7 MB in 5s (5012 kB/s)
Selecting previously unselected package libgpm2:amd64.
(Reading database ... 19289 files and directories currently installed.)
Preparing to unpack .../libgpm2_1.20.4-6.1_amd64.deb ...
Unpacking libgpm2:amd64 (1.20.4-6.1) ...
Preparing to unpack .../libssl1.0.0_1.0.2g-1ubuntu4.6_amd64.deb ...
Unpacking libssl1.0.0:amd64 (1.0.2g-1ubuntu4.6) over (1.0.2g-1ubuntu4.5) ...
Selecting previously unselected package vim-common.
Preparing to unpack .../vim-common_2%3a7.4.1689-3ubuntu1.2_amd64.deb ...
Unpacking vim-common (2:7.4.1689-3ubuntu1.2) ...
Preparing to unpack .../libicu55_55.1-7ubuntu0.1_amd64.deb ...
Unpacking libicu55:amd64 (55.1-7ubuntu0.1) over (55.1-7) ...
Selecting previously unselected package icu-devtools.
Preparing to unpack .../icu-devtools_55.1-7ubuntu0.1_amd64.deb ...
Unpacking icu-devtools (55.1-7ubuntu0.1) ...
Selecting previously unselected package libicu-dev:amd64.
Preparing to unpack .../libicu-dev_55.1-7ubuntu0.1_amd64.deb ...
Unpacking libicu-dev:amd64 (55.1-7ubuntu0.1) ...
Selecting previously unselected package libpython3.5:amd64.
Preparing to unpack .../libpython3.5_3.5.2-2ubuntu0~16.04.1_amd64.deb ...
Unpacking libpython3.5:amd64 (3.5.2-2ubuntu0~16.04.1) ...
Selecting previously unselected package zlib1g-dev:amd64.
Preparing to unpack .../zlib1g-dev_1%3a1.2.8.dfsg-2ubuntu4_amd64.deb ...
Unpacking zlib1g-dev:amd64 (1:1.2.8.dfsg-2ubuntu4) ...
Selecting previously unselected package libssl-dev:amd64.
Preparing to unpack .../libssl-dev_1.0.2g-1ubuntu4.6_amd64.deb ...
Unpacking libssl-dev:amd64 (1.0.2g-1ubuntu4.6) ...
Selecting previously unselected package vim-runtime.
Preparing to unpack .../vim-runtime_2%3a7.4.1689-3ubuntu1.2_all.deb ...
Adding 'diversion of /usr/share/vim/vim74/doc/help.txt to /usr/share/vim/vim74/doc/help.txt.vim-tiny by vim-runtime'
Adding 'diversion of /usr/share/vim/vim74/doc/tags to /usr/share/vim/vim74/doc/tags.vim-tiny by vim-runtime'
Unpacking vim-runtime (2:7.4.1689-3ubuntu1.2) ...
Selecting previously unselected package vim.
Preparing to unpack .../vim_2%3a7.4.1689-3ubuntu1.2_amd64.deb ...
Unpacking vim (2:7.4.1689-3ubuntu1.2) ...
Processing triggers for libc-bin (2.23-0ubuntu4) ...
Processing triggers for mime-support (3.59ubuntu1) ...
Setting up libgpm2:amd64 (1.20.4-6.1) ...
Setting up libssl1.0.0:amd64 (1.0.2g-1ubuntu4.6) ...
debconf: unable to initialize frontend: Dialog
debconf: (TERM is not set, so the dialog frontend is not usable.)
debconf: falling back to frontend: Readline
Setting up vim-common (2:7.4.1689-3ubuntu1.2) ...
Setting up libicu55:amd64 (55.1-7ubuntu0.1) ...
Setting up icu-devtools (55.1-7ubuntu0.1) ...
Setting up libicu-dev:amd64 (55.1-7ubuntu0.1) ...
Setting up libpython3.5:amd64 (3.5.2-2ubuntu0~16.04.1) ...
Setting up zlib1g-dev:amd64 (1:1.2.8.dfsg-2ubuntu4) ...
Setting up libssl-dev:amd64 (1.0.2g-1ubuntu4.6) ...
Setting up vim-runtime (2:7.4.1689-3ubuntu1.2) ...
Setting up vim (2:7.4.1689-3ubuntu1.2) ...
update-alternatives: using /usr/bin/vim.basic to provide /usr/bin/vim (vim) in auto mode
update-alternatives: using /usr/bin/vim.basic to provide /usr/bin/vimdiff (vimdiff) in auto mode
update-alternatives: using /usr/bin/vim.basic to provide /usr/bin/rvim (rvim) in auto mode
update-alternatives: using /usr/bin/vim.basic to provide /usr/bin/rview (rview) in auto mode
update-alternatives: using /usr/bin/vim.basic to provide /usr/bin/vi (vi) in auto mode
update-alternatives: using /usr/bin/vim.basic to provide /usr/bin/view (view) in auto mode
update-alternatives: using /usr/bin/vim.basic to provide /usr/bin/ex (ex) in auto mode
update-alternatives: using /usr/bin/vim.basic to provide /usr/bin/editor (editor) in auto mode
Processing triggers for libc-bin (2.23-0ubuntu4) ...
 ---> a94823f22e9c
Removing intermediate container f3fb98b35adf
Step 5/8 : COPY . /usr/local/plank
 ---> ce37742d59a8
Removing intermediate container 7a25de3d4cb0
Step 6/8 : RUN cd /usr/local/plank && swift build -c release
 ---> Running in d252c3cb5788
Compile Swift Module 'Core' (14 sources)
Compile Swift Module 'plank' (2 sources)
Linking ./.build/release/plank
 ---> 3f95d2cb3f39
Removing intermediate container d252c3cb5788
Step 7/8 : ENV plank_HOME /usr/local/plank
 ---> Running in ac4de46cba73
 ---> 678a01e4d282
Removing intermediate container ac4de46cba73
Step 8/8 : ENV PATH ${plank_HOME}/.build/release:${PATH}
 ---> Running in 220f32be7f15
 ---> 76adf601e56b
Removing intermediate container 220f32be7f15
Successfully built 76adf601e56b
```
